onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -16,6 +17,39 @@
 if typing.TYPE_CHECKING:
     import onnxruntime
 
+HAS_ONNX_REWRITTER = True
+try:
+    from onnxrewriter.rewriter.transformers import rewrite
+    from onnxrewriter.optimizer import optimize
```

Comment:
I could see a few problems:
(1) onnxrewriter is not available in other language C/C++/Nuget API. It make the result inconsistent across different language.
(2) If onnxrewriter has a bug, you have to uninstall it since no explicit option to disable it.
(3) It might not work well with large model (>2GB) since .SerializeToString() is used.

If onnxrewriter is generic enough, why not implement it inside onnxruntime with C++ optimizer?

Reply:
I am ok with whatever way onnx-rewriter should be called. I just use the only available way I can unblock llama with `onnxrt` dynamo backend (without this change, ORT is many times slower than inductor, and we will get no market share from PyTorch 2 features). I can add a flag to turn it on/off. Does it make sense?

For how this thing should ultimately implemented, please talk with @thiagocrepaldi. I guess exporter (i.e., onnxscript) will eventually include this optimization pass after it's matured.

Reply:
I've discussed this with Thiago. The current plan is that onnx-rewriter will be invoked by onnxscript and will be part of the exporter workflow. Having a flag is again not consistent between different language bindings. We don't want users to think that the Python bindings can give better perf with a flag, but that flag is not available for others. Can the llama model be unblocked by calling the rewriter separately?

Reply:
Nop. The only alternating solution I have in mind is to support `custom post-processing` pass in InferenceSession.